proxy/src/connection.rs

```diff
@@ -255,7 +258,7 @@ impl io::Read for Connection {
             // hold onto the allocated memory any longer. We won't peek
             // again.
             if peeked_len == len {
-                self.peek_buf = BytesMut::new();
+                self.peek_buf = Default::default();
```

Comment:
What is the functional difference between `BytesMut::new` and `<BytesMut  as Default>::default` in this case?

Reply:
> What is the functional difference between BytesMut::new and <BytesMut as Default>::default in this case?

There is no difference since `Default::default()` just calls `Self::new()`. I changed these to use `Default::default()` to help people see how, if we create an abstraction that works for both `Bytes` and `BytesMut`, we can make this code generic over those two, as a potential way to eliminate the duplication of logic.